### PR TITLE
Review doc for governance state [DON'T MERGE]

### PR DESCRIPTION
**Pull request by satran004**
Critical and high issues:
- [x] #822
- [x] #823
- [x] #824
- [x] #825 

 Medium issues:
- [x] #876 
- [x] #880 

Low issues:
- [x] #885

### CODE REVIEW

File: adr/governance-state-calculation-review.md [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+} else {
+    noStake = no.add(notVoted).add(noConfidence);
+}
+```
+
+**Impact:** For NO_CONFIDENCE actions, the yes ratio is slightly lower than it should be. This could prevent a valid NoConfidence motion from reaching its threshold.
+
+---
+
+### FINDING 2: DRep Voting Returns PASS When Total Yes+No Is Zero [MEDIUM]
```
Comment (Sotatek-HuyLe3a) [inline adr/governance-state-calculation-review.md]: github issue: #876

File: adr/governance-state-calculation-review.md [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+
+This is because Yaci Store computes expiry retroactively at epoch boundaries (from historical data), whereas the Haskell ledger updates expiry in real-time as DReps act. The V9 bonus is a workaround for the retrospective calculation approach.
+
+The TODO on line 145 (`//TODO: "<" or "<="`) indicates active uncertainty about boundary conditions.
+
+**Recommendation:** Add extensive test cases comparing Yaci Store DRep expiry calculations against known mainnet data for bootstrap-era DReps.
+
+---
+
+### FINDING 12: DRep Distribution - Stale Deregistration Check [LOW]
```
Comment (Sotatek-HuyLe3a) [inline adr/governance-state-calculation-review.md:290]: false positive. Cert type = 'UNREG_CERT' is also stored with type = 'STAKE_DEREGISTRATION'. No fix needed.

File: adr/governance-state-calculation-review.md [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+**Files:**
+- `aggregates/governance-rules/src/main/java/.../domain/RatificationContext.java:23-28`
+
+**Analysis:** `isOutOfLifecycle()` uses `> 1` and `isLastRatificationOpportunity()` uses `== 1`. At epoch `maxAllowedVotingEpoch + 1`, the proposal gets one last chance to ratify, matching the Haskell behavior where ratification is attempted BEFORE expiry is checked.
+
+**Status:** CORRECT.
+
+---
+
+### FINDING 9: ParameterChange - DRep Threshold Returns ZERO When Only Security Params Changed [HIGH]
```
Comment (Sotatek-HuyLe3a) [inline adr/governance-state-calculation-review.md:237]: fixed, gh issue: https://github.com/bloxbean/yaci-store/issues/825

File: adr/governance-state-calculation-review.md [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+- `aggregates/governance-rules/src/main/java/.../voting/spo/SPOVotingEvaluator.java:47-51`
+
+**Issue:**
+`isSPOVotingRequired()` returns `true` for ALL `PARAMETER_CHANGE_ACTION`, but SPOs only vote when security-relevant params are changed. The actual gate is in `ParameterChangeRatificationEvaluator` (line 49).
+
+**Status:** Not a functional bug due to the gate in the ratification evaluator. The `isSPOVotingRequired()` method is misleading but harmless.
+
+---
+
+### FINDING 11: DRep Expiry - V9 Bonus Logic Discrepancy [MEDIUM]
```
Comment (Sotatek-HuyLe3a) [inline adr/governance-state-calculation-review.md:274]: no fix needed

File: adr/governance-state-calculation-review.md [hunk 5]
```diff
@@ -0,0 +1,546 @@
...
+
+**Impact:** During bootstrap, NoConfidence, UpdateCommittee, NewConstitution, and TreasuryWithdrawals may fail because DRep voting returns `INSUFFICIENT_DATA` instead of `PASS_THRESHOLD`. However, during bootstrap on mainnet, some of these action types may be restricted by the GOV rule (not all actions allowed during bootstrap), limiting the practical impact.
+
+**Recommendation:** Either:
+1. Make `DRepVotingEvaluator` bootstrap-aware (return `PASS_THRESHOLD` during bootstrap), OR
+2. Each ratification evaluator should treat `INSUFFICIENT_DATA` as pass during bootstrap for DRep votes.
+
+---
+
+### FINDING 19: Proposal Expiry - `maxAllowedVotingEpoch` Source Parameters [INFO]
```
Comment (Sotatek-HuyLe3a) [inline adr/governance-state-calculation-review.md:429]: No fix needed. In yaci -store, maxAllowedVotingEpoch = proposal submission epoch + gov action lifetime from the pp at the time of proposal submission

File: adr/governance-state-calculation-review.md [hunk 6]
```diff
@@ -0,0 +1,546 @@
...
+
+**Problem:** `minFeeRefScriptCostPerByte` is listed ONLY in the SECURITY group check (`ProtocolParamUtil.java:39-42`). It does NOT appear in any regular group (ECONOMIC, NETWORK, TECHNICAL, GOVERNANCE). In the Haskell ledger, it has `PPGroups EconomicGroup SecurityGroup`.
+
+If a ParameterChange modifies only `minFeeRefScriptCostPerByte`, the groups would be `[SECURITY]`. After filtering: empty. DRep threshold = 0 (auto-pass). **The correct threshold should be `dvtPPEconomicGroup`.**
+
+**Recommendation:** Add `minFeeRefScriptCostPerByte` to the ECONOMIC group check in `ProtocolParamUtil`. Verify all SECURITY params also have a regular group classification matching the Haskell `PPGroups` tags.
+
+---
+
+### FINDING 10: ParameterChange - SPO Voting Required Check Overly Broad [LOW]
```
Comment (Sotatek-HuyLe3a) [inline adr/governance-state-calculation-review.md:262]: No fix needed

File: adr/governance-state-calculation-review.md [hunk 7]
```diff
@@ -0,0 +1,546 @@
...
+  | otherwise = False
+```
+
+When `totalAbstain == totalStake`, the Haskell code does NOT auto-pass. The ratio would be `0 / (total - total)` = `0/0`, which evaluates as 0 in Haskell's `Rational` type. So `0 >= threshold` is False for non-zero thresholds.
+
+**Recommendation:** When `totalStake == totalAbstain`, check if the threshold is zero (minBound). Only auto-pass if threshold is zero. Otherwise return `NOT_PASS_THRESHOLD`.
+
+---
+
+### FINDING 4: Committee Voting - Do-Not-Vote as NO May Double-Count [LOW]
```
Comment (Sotatek-HuyLe3a) [inline adr/governance-state-calculation-review.md:136]: Fixed, gh issue: https://github.com/bloxbean/yaci-store/issues/885
